File: src/backend/app/schemas/trajectory.py
# app/schemas/trajectory.py
from pydantic import BaseModel

class Position(BaseModel):
    international_designator: str
    az: float
    alt: float

class Trajectory(BaseModel):
    timestamp: str
    positions: list[Position]

class TrajectoryResponse(BaseModel):
    location_name: str
    trajectories: list[Trajectory]

File: src/frontend/src/types/trajectory.ts
// src/types/trajectory.ts
export interface SatPosition {
  international_designator: string;
  az: number;
  alt: number;
}

export interface Trajectory {
  timestamp: string;
  positions: SatPosition[];
}

export interface TrajectoryResponse {
  trajectories: Trajectory[];
}
